src/main.h

```diff
@@ -191,8 +191,10 @@ static const uint64_t nMinDiskSpace = 52428800;
 /** Pruning-related variables and constants */
 /** True if any block files have ever been pruned. */
 extern bool fHavePruned;
-/** True if we're running in -prune mode. */
-extern bool fPruneMode;
+enum PruneMode {PRUNE_NONE = 0, PRUNE_AUTO, PRUNE_MANUAL };
```

Comment:
Let's use C++11 scoped enums in new code
```
enum struct PruneMode { NONE=0, AUTO, MANUAL };
```
Then refer to `PruneMode::NONE` etc.